src/QueryBuilder/Patterns/Scope.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Osi\QueryBuilder\Patterns;
+
+use InvalidArgumentException;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Validator;
+
+/**
+ * Class Scope
+ * @package Osi\QueryBuilder\Patterns
+ */
+class Scope implements Pattern
+{
+    const RULES = [
+        'pattern' => 'required|in:SCOPE',
+        'not'     => 'boolean'
+    ];
+
+    /**
+     * @param string $field
+     * @param array $params
+     * @param Builder $queryBuilder
+     * @return void
+     * @throws InvalidArgumentException
+     */
```

Comment:
You can remove the doc block here since everything is with types now.

Reply:
I want to be consistent with public packages and leave all comment because there are methods can't remove comments. After PHP 8 we will remove all comments